thirdparty/etcpak/patches/0001-Make-mingw-deal-with-intrinsics-better-and-narrowing.patch

```diff
@@ -0,0 +1,199 @@
+From 6859e3407df2520c5b97207ec42584e5c0cdcc57 Mon Sep 17 00:00:00 2001
+From: "K. S. Ernest (iFire) Lee" <ernest.lee@chibifire.com>
+Date: Tue, 6 Apr 2021 21:58:02 -0700
+Subject: [PATCH] Make mingw deal with intrinsics better and narrowing
+ conversion.
+
+---
+ thirdparty/etcpak/BlockData.cpp  | 63 ++++++++++++++++++--------------
+ thirdparty/etcpak/ProcessRGB.cpp | 32 ++++++++++------
+ thirdparty/etcpak/System.cpp     |  4 +-
+ 3 files changed, 59 insertions(+), 40 deletions(-)
+
+diff --git a/thirdparty/etcpak/BlockData.cpp b/thirdparty/etcpak/BlockData.cpp
+index a2cd032c5bf..0ab193387b2 100644
+--- a/thirdparty/etcpak/BlockData.cpp
++++ b/thirdparty/etcpak/BlockData.cpp
+@@ -16,7 +16,7 @@
+ #endif
+ 
+ #ifdef __SSE4_1__
+-#  ifdef _MSC_VER
++#  if defined(_MSC_VER) && !defined(__MINGW32__)
+ #    include <intrin.h>
+ #    include <Windows.h>
+ #    define _bswap(x) _byteswap_ulong(x)
+@@ -24,17 +24,26 @@
+ #  else
+ #    include <x86intrin.h>
+ #  endif
+-#else
+-#  ifndef _MSC_VER
+-#    include <byteswap.h>
+-#    define _bswap(x) bswap_32(x)
+-#    define _bswap64(x) bswap_64(x)
+-#  endif
+ #endif
+-
+-#ifndef _bswap
+-#  define _bswap(x) __builtin_bswap32(x)
+-#  define _bswap64(x) __builtin_bswap64(x)
++#if !defined(_bswap) && !defined(_bswap64)
++// Copyright 2020 github user jtbr, Released under MIT license
++// https://gist.github.com/jtbr/7a43e6281e6cca353b33ee501421860c
++static inline uint32_t _bswap(uint32_t x) {
```

Comment:
For the reference, we have our own definitions for bswap in Godot which are probably trivial enough to be considered public domain, and could be used without adding another copyright and license to etcpak:
https://github.com/godotengine/godot/blob/8b6e3d63750db5eab94b5d4e5dc182a678c1e455/core/typedefs.h#L210-L230

I didn't check if they're strictly equivalent.